piptools/utils.py

```diff
@@ -514,37 +508,6 @@ def copy_install_requirement(
     return ireq
 
 
-class PackageMetadata(Protocol):
-    def get_all(self, name: str, failobj: _T = ...) -> list[str] | _T:
-        ...
```

Comment:
Perhaps, restore this code to address this https://github.com/jazzband/pip-tools/pull/1681#discussion_r1299370048?

Reply:
I tried it and it doesn’t work because in some places the type hints include overloads.

Reply:
Done, see other thread.

Reply:
Thanks!